media/js/base/protocol/init-navigation.js

```diff
@@ -41,6 +41,62 @@
         }
     }
 
+    var nav = document.querySelector('.mzp-c-navigation');
+    var fxaButton = document.querySelector('.mzp-c-navigation .c-navigation-fxa-cta');
+
+    if (fxaButton) {
+        var fxaButtonAltHref = fxaButton.getAttribute('data-alt-href');
+    }
+
+    function showFxAButton() {
+        nav.classList.add('show-fxa-button');
+    }
+
+    function meetsRequirements() {
+        if (typeof Mozilla.Client === 'undefined') {
+            return false;
+        }
+
+        // User should be on Firefox desktop, nav should be present on page, and the FxA button should exist.
+        if (!Mozilla.Client.isFirefoxDesktop || !nav || !fxaButton) {
+            return false;
+        }
+
+        var userMajorVersion = Mozilla.Client._getFirefoxMajorVersion();
+        var latestMajorVersion = parseInt(document.documentElement.getAttribute('data-latest-firefox'), 10);
+
+        if (!userMajorVersion || !latestMajorVersion) {
+            return false;
+        }
+
+        // User should be on Firefox Quantum or greater.
+        return userMajorVersion >= 57;
```

Comment:
I know the original ask for this was to show the button for FF >=  57, but there could be a simpler, more performant way to cut down on all this logic (perhaps)?

1. Default the button `href` to `/firefox/accounts/`.
2. Use the `is-firefox` class that gets added in `site.js` to determine showing the button (no additional JS required, just some CSS).
3. Then use UITour here to upgrade the button `href` to point to `acconts.firefox.com` if not already signed in.

This would mean the button would get shown to all Firefoxes, and not just desktop, but the logic to handle all those cases should be built into the /accounts/ page (I think?). 

Just an idea, but it does change some parameters. And what you have works, so totally feel free to say no 🙂

Reply:
This is slightly complicated by the `fxa_link_fragment` macro which includes the firefox.com.cn link for China. We could update that macro, but then it's also in use on other pages and I'm worried about blowing up the scope of this and delaying it even further 😟 

But I think we could definitely just show it to all Firefoxes and not worry about the version and all the other checks. Do we need to still check that it's desktop and not mobile? What happens if you get the button on mobile? I guess I should try that...

Reply:
Good point, let's not creep the scope on this further than it needs to be. We definitely shouldn't be pointing mobile users to accounts.firefox.com though. The FxA team we're pretty specific on not doing this when we ran the experiment, as that page is for desktop users only.